### PR TITLE
Improve 'site user' to allow anyone to use it and impose restrictions in code

### PR DESCRIPTION
**Pull request by mvangoor**

### CODE REVIEW

**Comment by mvangoor** [comment]
The wrappers can safely be ignore, but the merge conflict for UserManagementHandler.java is bit more meh ..

**Comment by k-2-r** [comment]
Please rebase your branch and fix the PR to prevent conflict :)

**Comment by mvangoor** [comment]
This has been rebased and force pushed. Please check

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java [hunk 1]
```diff
@@ -1817,12 +1817,27 @@ public CommandResponse doSITE_USER(CommandRequest request) throws ImproperUsageE
 
         Session session = request.getSession();
 
-        if (request.getUser() == null) {
-            return StandardCommandManager.genericResponse("RESPONSE_530_ACCESS_DENIED");
+        // This command can be accessed by anyone, we impose restrictions here to ensure no data is leaked we do not want to be leaked
+        boolean hasPermission = false;
+
+        // Basic check
+        if (request.getUser() != null) {
```
Comment (k-2-r) [inline core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1824]: Maybe is a naming variable problem but looks like :
> myUser is the user requested by the command for now. But it should be named requestedUser
> reqUser is the current authenticated user that ask for other user information. Should be named currentUser

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java [hunk 2]
```diff
@@ -1817,12 +1817,27 @@ public CommandResponse doSITE_USER(CommandRequest request) throws ImproperUsageE
 
         Session session = request.getSession();
 
-        if (request.getUser() == null) {
```
Comment (k-2-r) [inline core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1820]: Should not be removed. If the requestedUser is null, the command must be aborted. ACCESS DENIED is send but i think to not possibly disclose that the user exists.

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java [hunk 3]
```diff
@@ -1817,12 +1817,27 @@ public CommandResponse doSITE_USER(CommandRequest request) throws ImproperUsageE
...
-            return StandardCommandManager.genericResponse("RESPONSE_530_ACCESS_DENIED");
+        // This command can be accessed by anyone, we impose restrictions here to ensure no data is leaked we do not want to be leaked
+        boolean hasPermission = false;
+
+        // Basic check
+        if (request.getUser() != null) {
+            User reqUser = session.getUserNull(request.getUser());
+            if (reqUser.equals(myUser)) {
+                hasPermission = true;
+            } else if(reqUser.isAdmin()) {
```
Comment (k-2-r) [inline core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1828]: Maybe i dont understand but its look like if the user you want info is an admin you allow the data fetch. wanted?

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java [hunk 4]
```diff
@@ -1817,12 +1817,27 @@ public CommandResponse doSITE_USER(CommandRequest request) throws ImproperUsageE
...
+                hasPermission = true;
+            } else {
+                if (reqUser.isGroupAdmin()) {
+                    if (myUser.isMemberOf(reqUser.getGroup())) {
+                        hasPermission = true;
+                    }
+                }
+            }
         }
```
Comment (k-2-r) [inline core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1837]: Could be interesting to have a clearer list of condition.
boolean myself = reqUser.equals(myUser);
boolean adminUser = reqUser.isAdmin();
boolean askedUserInSameGroupAsMyself = ....
And then an if with or conditions to grant or not the command.

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java (hunk 2 above)
Comment (mvangoor) [reply core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1820]: You are correct. Fixing

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java (hunk 3 above)
Comment (mvangoor) [reply core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1828]: Yes, the function is isAdmin() is not really true to the actual meaning. This checks if the user is part of group 'siteop' which means full admin rights. So yes this should be there

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java (hunk 1 above)
Comment (mvangoor) [reply core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1824]: Changed the naming

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java (hunk 3 above)
Comment (k-2-r) [reply core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1828]: So anyone can access to admin user information? Its really what is expected?

File: core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java (hunk 4 above)
Comment (mvangoor) [reply core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java:1845]: I added comments to showcase why it is allowing. I do not really see the benefit of doing this another way.

**Pull request merged**